zipper/helper/requests.go

```diff
@@ -5,17 +5,114 @@ import (
 	"context"
 	"io"
 	"io/ioutil"
+	"net"
 	"net/http"
 	"net/url"
 	"sync/atomic"
 
 	"github.com/ansel1/merry"
 	"github.com/go-graphite/carbonapi/limiter"
+	"github.com/go-graphite/carbonapi/pkg/parser"
 	util "github.com/go-graphite/carbonapi/util/ctx"
 	"github.com/go-graphite/carbonapi/zipper/types"
 	"go.uber.org/zap"
 )
 
+func HttpCode(err error) int {
+	if err == nil {
+		return http.StatusOK
+	}
+	if _, ok := err.(*net.OpError); ok {
+		return http.StatusServiceUnavailable
+	}
+	if urlErr, ok := err.(*url.Error); ok {
+		if _, ok = urlErr.Err.(*net.OpError); ok {
+			return http.StatusServiceUnavailable
+		}
+	}
+	return http.StatusInternalServerError
+}
+
+func MergeHttpErrors(errors []merry.Error) (int, []string) {
+	returnCode := http.StatusNotFound
+	errMsgs := make([]string, 0)
+	for _, err := range errors {
+		var code int
+		c := merry.RootCause(err)
+		if c == nil {
+			c = err
+		}
+		code = merry.HTTPCode(c)
+
+		if merry.HTTPCode(c) == 404 || merry.Is(c, parser.ErrSeriesDoesNotExist) {
```

Comment:
even though current code doesn't have consistency across codes, but I'd suggest to try to replace all hardcoded codes as constants from net/http, e.x. 404 -> http.StatusNotFound